radix-engine/src/system/system_callback.rs

```diff
@@ -19,7 +20,8 @@ use radix_engine_interface::blueprints::resource::{
 };
 use radix_engine_interface::schema::BlueprintSchema;
 
-fn validate_input(
+fn validate_input<'a, Y: KernelApi<SystemConfig<V>>, V: SystemCallbackObject>(
+    mut upstream: SystemUpstream<'a, Y, V>,
```

Comment:
This is rather weird abstraction. Is this only needed for the `get_node_type_info` api? Probably better to create `SystemDownstream` object earlier in `invoke_upstream`

Reply:
`SystemDownstream` was invented as a simple container type, so that I can implement traits for it. 

I will rename it to something else, so not to confuse with real system downstream.